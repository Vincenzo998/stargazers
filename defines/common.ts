export const USER = "user";
export const REPO = "repo";
