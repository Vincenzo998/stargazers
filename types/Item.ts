export type Item = { id: string; avatar_url: string; login: string };
