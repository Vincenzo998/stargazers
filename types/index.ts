export * from "./ListItemProps";
export * from "./Item";
export * from "./Info";
