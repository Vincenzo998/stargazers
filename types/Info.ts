// defines
import { REPO, USER } from "../defines";

export type Info = { [USER]: string; [REPO]: string };
