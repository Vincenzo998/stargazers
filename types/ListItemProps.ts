import { Item } from "./Item";

export type ListItemProps = { item: Item; index: number };
