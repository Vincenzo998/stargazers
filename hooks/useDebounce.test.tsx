import { renderHook } from "@testing-library/react-hooks";

// screens
import { useDebounce } from "./useDebounce";

// types
import { Info } from "../types";

// defines
import { REPO, USER } from "../defines";

test("useDebounce is invoked without crashing", () => {
  const info: Info = {
    [USER]: "user",
    [REPO]: "user-repo",
  };

  const {
    result: { current },
  } = renderHook(() => useDebounce<Info>(info, 500));

  expect(current).toMatchSnapshot();
});
