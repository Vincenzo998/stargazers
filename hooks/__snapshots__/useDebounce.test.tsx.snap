// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useDebounce is invoked without crashing 1`] = `
{
  "repo": "user-repo",
  "user": "user",
}
`;
